src/actions/router.js

```diff
@@ -10,16 +10,18 @@ import * as history from '../app/history'
 import { setChannel, openPm, openChannel } from './'
 
 export function goTo(pathOrUrl, options = {}) {
-  return dispatch => {
+  return (dispatch, getState) => {
     dispatch({
       type: types.GO_TO,
       payload: options,
     })
 
+    const currentChannel = channelSelector(getState())
+
     openUrl(pathOrUrl, {
       serviceUrl: conf.server.serviceUrl,
       mode: getMode(conf),
-      currChannel: conf.channelId,
+      currChannel: conf.channelId || currentChannel.id,
```

Comment:
We should cover the case that currentChannel is undefined!

Reply:
in `openUrl` function there is only one place (https://github.com/ubergrape/grape-web/blob/master/src/x-platform/open-url/index.js#L67) where we using `currChannel ` from `options`. It's a case when a link (without `messageId` in url) leads to the same channel. For example - `@group` mention. This mentions we can meet only in message history, and in message search. Am I right in this statement? And if we don't have the current channel it means we don't have channels at all, and obviously, any messages where we can meet this type of mention. Maybe I got your message wrong, and don't see case where `currentChannel` can be `undefined` in this case

Reply:
For me it's mostly about defensive programming. I'm not too much concerned about how openUrl is used currently, but if we use it in the future in a different context we might automatically create a bug. Or let's say we change the existing code and there can be a case were currentChannel is undefined.

Of course we have to take certain assumptions for granted, but minimising those should help us to make the whole application less fragile.